### PR TITLE
Migrate to null safety

### PR DESCRIPTION
**Pull request by miDeb**
Since this is a very simple package, migrating to null safety only requires updating the sdk constraints!

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -15,7 +15,7 @@ description: Compute hashCode correctly for your objects
 homepage: https://github.com/QuiverDart/quiver_hashcode
 
 environment:
-  sdk: '>=2.0.0-dev.61 <3.0.0'
+  sdk: '>=2.12.0 <3.0.0'
 
 dev_dependencies:
   test: '^1.0.0'
```
Comment (cbracken) [inline pubspec.yaml:21]: Looks like this likely needs updating to `^1.16.0-nullsafety.8`

**Review by cbracken — COMMENTED** [review]
Hi @miDeb thanks for the PR!

The quiver sub-packages under the QuiverDart org are pretty much unmaintained at this point and we recommend people use the main [Quiver](http://github.com/google/quiver-dart) package, which is already nnbd-ready. This won't change the size of the resultant app since Dart's tree-shaking compiler eliminates any code that's not called from your app, both in VM-based/Flutter cases and on the web. 

The latest NNBD-ready version of that package is 3.0.0-nullsafety.3.

I actually thought we'd updated the READMEs of the subpackages here to point people to the main package, but apparently we didn't. I'm happy to take this PR, but it'll likely need an update to the test package constraint.

**Comment by miDeb** [comment]
Thanks @cbracken, I've updated the test dependency (though nothing complained about the old version, so not sure if it was really required). I guess I'll propose to the other package I was trying to migrate to switch from this package to quiver. Thank you anyways.

**Comment by cbracken** [comment]
Thanks and sorry for the slow turnaround here. v3.0.0 is published and we've updated the readme and the pub.dev page to point at the main package.

**Pull request closed without merging**